README.md

```diff
@@ -22,6 +22,132 @@ By group 36 of the course "Software Praktikum"@UZH in FS 22.
+- SonarQube (static code analysis and technical code quality evaluation)
+
+### Spring Framework, Spring Boot
+![spring](./doc/img/spring.png) 
+
+#### Spring Boot
+Spring Boot helps with the creation of stand-alone Spring based applications.
+
+#### Spring Data
+The rapid api development architecture provided by spring-data-rest was used to develop the api.
+It provides an easy way to build REST web services on top of the data repositories. The
+PagingAndSortingRepositories provide request handling, deserialization, crud on the database,
+serialization and rendering of errors. Validation was implemented with Bean validations and if
+necessary with Spring event handlers. Side effects were also done with Spring event handlers.
+
+### Hibernate
+![hibernate](./doc/img/hibernate.png)
+
+The data was stored in a hibernate database which offers an object relational mapping of the data.
+This means object oriented entities can be stored in a relational database.
+
+### JPA
+Jakarta Persistence API is the API for storing, retrieving, updating and deleting relational database
+entries in an object oriented context.
+
+### Rapid API
+![rapid api](./doc/img/rapid_api.png)
+
+??? --> Frontend?
+
+### PlantUML
+![plant](./doc/img/plant_uml.png)
+
+We used the PlantUML plain text language to create our UML diagrams.
+
+### Git, Github
+The versioning of our source code and other files was maintained by Git ont the GitHub platform.
+We used the GitHubs Actions feature for automated build, test and deployment of the software.
+
+## High-level components
+### Database
+We persisted all the entities used for the game. We decided to do so, because Spring Data REST offered
+various possibilities to implement validations and side effects in an easy way.
+- Bean validation (uniqueness, not null, format etc.)
+- Ensure referential integrity
+- Locking data for updates (ensure consistency)
+- Side effects: triggering some additional (game) logic before or after creating, updating or deleting an entity
+
+### Game Logic (side effects)
+The game logic was mainly part of the EventHandlers. They implemented so-called side effects which were triggered
+by certain database interactions. Spring ApplicationListener listens to creating, saving and deleting
+events and executes then the logic with the corresponding annotation (e.g. a method with the `@HandleAfterSave`
+annotation is executed after the update of an entity).
+
+### Validations
+Validations are implemented in several places. Format validations are implemented on entities level
+by bean validations. Some Validator classes dedicated to specific entities do further game validations.
+The CardValidator class for example ensures that a player can not play two cards in the same round.
+
+Validation errors as well as technical exceptions from Java or Hibernate are catched and transformed 
+in appropriate HTTP status codes with informal exception messages to prevent technical errors in the
+frontend.
```

Comment:
Should we create a table with validation errors and respective http status codes?